lib/cocoapods/target/pod_target.rb

```diff
@@ -61,7 +68,7 @@ def initialize(specs, target_definitions, sandbox, scoped = false)
     #
     def scoped
       target_definitions.map do |target_definition|
-        PodTarget.new(specs, [target_definition], sandbox, true).tap do |target|
+        PodTarget.new(specs, [target_definition], sandbox, target_definition.label).tap do |target|
```

Comment:
You can't reproduce with that, what the analyzer archives. But the label of the target definition is always a satisfying denominator. This is used in the fallback case and as helper for our tests.